### PR TITLE
Implement FromStr for OsString

### PR DESCRIPTION
**Pull request by glandium**

### CODE REVIEW

File: src/libstd/ffi/os_str.rs [hunk 1]
```diff
@@ -1174,6 +1175,15 @@ impl AsInner<Slice> for OsStr {
     }
 }
 
+#[stable(feature = "osstring_from_str", since = "1.45.0")]
+impl FromStr for OsString {
```
Comment (lzutao) [inline src/libstd/ffi/os_str.rs:1179]: What about OsStr?

Comment (CryZe) [reply src/libstd/ffi/os_str.rs:1179]: FromStr can't borrow.

Comment (lzutao) [reply src/libstd/ffi/os_str.rs:1179]: I see it now. Thanks for the clarification!

**Comment by sfackler** [comment]
@rfcbot fcp merge

**Pull request merged**